test/src/main/java/org/apache/accumulo/test/functional/CreateInitialSplitsIT.java

```diff
@@ -0,0 +1,218 @@
+import org.apache.accumulo.core.util.TextUtil;
+import org.apache.accumulo.harness.AccumuloClusterHarness;
+import org.apache.accumulo.minicluster.MemoryUnit;
+import org.apache.accumulo.minicluster.ServerType;
+import org.apache.accumulo.minicluster.impl.MiniAccumuloConfigImpl;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.RawLocalFileSystem;
+import org.apache.hadoop.io.Text;
+import org.junit.Before;
+import org.junit.Test;
+
+@SuppressWarnings("SpellCheckingInspection")
+public class CreateInitialSplitsIT extends AccumuloClusterHarness {
+
+  private Connector connector;
+  private String tableName;
+
+  @Override
+  public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration conf) {
+    cfg.setMemory(ServerType.TABLET_SERVER, 128 * 4, MemoryUnit.MEGABYTE);
+
+    // use raw local file system
+    conf.set("fs.file.impl", RawLocalFileSystem.class.getName());
+  }
+
+  @Override
+  protected int defaultTimeoutSeconds() {
+    return 2 * 60;
+  }
+
+  @Before
+  public void setupInitialSplits() {
+    connector = getConnector();
+  }
+
+  /**
+   * Verify normal table creation did not get broken.
+   */
+  @Test
+  public void testCreateTableWithNoSplits()
+      throws TableExistsException, AccumuloSecurityException, AccumuloException {
+    tableName = getUniqueNames(1)[0];
+    connector.tableOperations().create(tableName);
+    assertTrue(connector.tableOperations().exists(tableName));
+  }
+
+  /**
+   * Create initial splits by providing splits from a Java Collection.
+   */
+  @Test
+  public void testCreateInitialSplits() throws TableExistsException, AccumuloSecurityException,
+      AccumuloException, TableNotFoundException {
+    tableName = getUniqueNames(1)[0];
+    SortedSet<Text> expectedSplits = generateNonBinarySplits(3000, 32);
+    NewTableConfiguration ntc = new NewTableConfiguration().withSplits(expectedSplits);
+    assertFalse(connector.tableOperations().exists(tableName));
+    connector.tableOperations().create(tableName, ntc);
+    assertTrue(connector.tableOperations().exists(tableName));
+    Collection<Text> createdSplits = connector.tableOperations().listSplits(tableName);
+    assertEquals(expectedSplits, new TreeSet<>(createdSplits));
+  }
+
+  @Test
+  public void testCreateInitialSplitsWithEncodedSplits() throws TableExistsException,
+      AccumuloSecurityException, AccumuloException, TableNotFoundException {
+    tableName = getUniqueNames(1)[0];
+    SortedSet<Text> expectedSplits = generateNonBinarySplits(3000, 32, true);
+    NewTableConfiguration ntc = new NewTableConfiguration().withSplits(expectedSplits);
+    assertFalse(connector.tableOperations().exists(tableName));
+    connector.tableOperations().create(tableName, ntc);
+    assertTrue(connector.tableOperations().exists(tableName));
+    Collection<Text> createdSplits = connector.tableOperations().listSplits(tableName);
+    assertEquals(expectedSplits, new TreeSet<>(createdSplits));
```

Comment:
It would also be nice to do the following to ensure a table created in this way does not have any quirks.

 1. Write some data to multiple tablets
 2. Verify data
 3. Compact table
 4. Verify data
 5. Delete table.